primitives/messages/src/lib.rs

```diff
@@ -89,6 +89,33 @@ impl TypeId for LaneId {
 	const TYPE_ID: [u8; 4] = *b"blan";
 }
 
+/// Lane direction.
+#[derive(Copy, Clone, Debug, Decode, Encode, Eq, PartialEq, TypeInfo, MaxEncodedLen)]
+pub enum LaneDirection {
```

Comment:
Just sharing my thoughts - maybe some may be implemented:
1) we are only using it (`LaneDirection` and `FullLaneId`) in the rewards/refund contest, so maybe move it to the `bp-relayers` crate? We don't actually need to pass it to the `DeliveryConfirmationPayments::pay_rewards()`, because, as name says, it is called during delivery confirmation transaction, so the direction would always be the `Out`. And you can build this `FullLaneId` in the implementation (`DeliveryConfirmationPaymentsAdapter`) when calling `register_relayer_rewards`
2) (very questionable) implementing (1) would allow us to use some reward/refund specific names (and comments) for this enum and struct. Since `FullLaneId` is used to compute account id that is used to pay rewards, then maybe rename `FullLaneId` to `RewardsAccountParams` and `LaneDirection` to `RewardsSource { ThisChainPaysRewards, BridgedChainPaysRewards }`. Why: the same problem isn't trivial (with who's paying for what), so imo we need to use all means to make it visible to ourselves and our crate users (at least for relayers)
3) in any case, let's add some docs (maybe to this srtuct), explaining why we are using bridged chain id and lane direction to compute rewards-source-account. Also would be great to have some extended docs for the `claim_rewards`, so that relayers know exactly what they need to pass to the call from its docs. Like: "if you have been submitting messages proof to this chain - pass <this>, if you have been submitting messages proof to the other chain - pass <this>, if you have been submitting messages **delivery** proof to this chain - pass <this>, ..."

Reply:
All the suggestions look good. Thanks ! Tried to implement all of them.